Tests/DeviceDetectorTest.php

```diff
@@ -261,6 +261,89 @@ public function getFixtures(): array
         return $fixtures;
     }
 
+    /**
+     * @dataProvider getFixturesClient
+     */
+    public function testParseClient(array $fixtureData): void
+    {
+        $ua          = $fixtureData['user_agent'];
+        $clientHints = !empty($fixtureData['headers']) ? ClientHints::factory($fixtureData['headers']) : null;
+
+        AbstractDeviceParser::setVersionTruncation(AbstractDeviceParser::VERSION_TRUNCATION_NONE);
+
+        try {
+            $uaInfo = DeviceDetector::getInfoFromUserAgent($ua, $clientHints);
+        } catch (\Exception $exception) {
+            throw new \Exception(
+                \sprintf('Error: %s from useragent %s', $exception->getMessage(), $ua),
+                $exception->getCode(),
+                $exception
+            );
+        }
+
```

Comment:
it will be easier to find the problem

```php        
        $messageError = \sprintf("Fixture:\n%s\nResult:\n%s",
            \Spyc::YAMLDump($fixtureData, 2, 0),
            \Spyc::YAMLDump($uaInfo, 2, 0)
        );

       unset($fixtureData['headers']); // ignore headers in result
       unset($fixtureData['client']['family']);

        $this->assertArrayNotHasKey('bot', $uaInfo, $messageError);
        $this->assertEquals($fixtureData['client'], $uaInfo['client'], $messageError);
```